### PR TITLE
ICMSLST-1429 Merge

### PR DESCRIPTION
**Pull request by davidwtbuxton**
- Merge upstream changes.

The merge had conflicts, most of which were in Python dependency versions listed in the Pipfile.lock.

The other significant conflict was in the mail/library/builders.py:build_email_message function, which was changed to use Django's mail classes.

- Fix SMTP-related tests.
    
The tests broke because we merged changes from upstream, which uses a single
MailServer class to handle POP3 and SMTP. But in icms-hmrc the SMTP stuff
has been moved to a Django mail back-end. Changing the new code to use
the default SMTP connection fixes the tests.

### CODE REVIEW

**Comment by davidwtbuxton** [comment]
Relevant changes made specifically in this repo are

https://github.com/uktrade/icms-hmrc/pull/8/commits/4ee9e2fef00f5b4094a91c3095a6c115bc83baab

and 

https://github.com/uktrade/icms-hmrc/pull/8/commits/912607c6390c7dd5120c9399d436f681c2649879

File: mail/management/commands/resend_email.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+                logger.error(
+                    "Mail is expected to be in 'reply_pending' status but current status is {0}".format(mail.status)
+                )
+                return
+        # this does happen occassionally
+        elif mail.extract_type == ExtractTypeEnum.LICENCE_REPLY:
+            destination = SourceEnum.SPIRE
+            if mail.status != ReceptionStatusEnum.REPLY_SENT:
+                logger.error(
+                    f"Mail is expected to be in 'reply_sent' status but current status is {0}".format(mail.status)
```
Comment (mikeylockwood) [inline mail/management/commands/resend_email.py:109]: Why are there `f-strings` in here using `.format`?

File: mail/libraries/helpers.py [hunk 2]
```diff
@@ -328,11 +336,13 @@ def get_action(reference) -> str:
 def get_country_id(country):
```
Comment (mikeylockwood) [inline mail/libraries/helpers.py:336]: Probably could do with a docstring on this function

File: mail/libraries/lite_to_edifact_converter.py [hunk 3]
```diff
@@ -103,7 +103,8 @@ def licences_to_edifact(licences: QuerySet, run_number: int) -> str:
             elif licence_payload.get("type") in LicenceTypeEnum.STANDARD_LICENCES:
                 if licence_payload.get("end_user"):
                     # In the absence of country_group or countries use country of End user
-                    country_id = licence_payload.get("end_user").get("address").get("country").get("id")
+                    country = licence_payload.get("end_user").get("address").get("country")
```
Comment (mikeylockwood) [inline mail/libraries/lite_to_edifact_converter.py:106]: What's the point of using `.get` here? It'll just error anyway if the first two keys aren't in the dict

File: mail/management/commands/resend_email.py (hunk 1 above)
Comment (MattHolmes123) [reply mail/management/commands/resend_email.py:109]: I don't think it works either.

Comment (MattHolmes123) [reply mail/management/commands/resend_email.py:109]: >>> f"this is broken {0}".format("hey")
'this is broken 0'

Comment (davidwtbuxton) [reply mail/management/commands/resend_email.py:109]: What approach to take here?

I can either fix it in this merge. Or merge as is, make a PR upstream. Or wait for it to be fixed upstream, merge changes then.

**Review by MattHolmes123 — APPROVED** [review]

**Review by mikeylockwood — APPROVED** [review]
Merged issues to be fixed upstream

**Pull request merged**